test/higher_order_ops/test_invoke_subgraph.py

```diff
@@ -24,7 +26,9 @@ def gn(x, y):
             return (torch.mul(x, y),)
 
         def fn(x, y):
-            return invoke_subgraph(gn, None, (x, y))[0]
+            return invoke_subgraph(FunctionWithNoFreeVars(gn), (x, y), identifier=None)[
```

Comment:
PrimHOPBase requires that functions passed to it that are not subgraphs be annotated as having no free variables. We can work around this restriction in the user-facing API for invoke_subgraph (perhaps by always turning on torch.compile)